python/dg_tools/leg_impedance_control/quad_leg_impedance_controller.py

```diff
@@ -165,36 +165,44 @@ def __init__(self, robot, ViconClientEntity, client_name = "vicon_client" , vico
         self.EntityName = EntityName
         self.client_name = client_name
         self.host_name_quadruped = vicon_ip
-        self.robot_vicon_name = "quadruped"
+        self.init_robot_properties()
 
-        self.vicon_client = ViconClientEntity(self.client_name)
+        self.vicon_client = ViconClientEntity('vicon_' + self.client_name)
         self.vicon_client.connect_to_vicon(self.host_name_quadruped)
         self.vicon_client.displaySignals()
 
         self.vicon_client.add_object_to_track("{}/{}".format(self.robot_vicon_name, self.robot_vicon_name))
 
         try:
             ## comment out if running on real robot
-            self.vicon_client.robot_wrapper(robot)
+            self.vicon_client.robot_wrapper(robot, self.robot_vicon_name)
         except:
             print("not in simulation")
 
 
-        self.robot.add_ros_and_trace(self.client_name, self.robot_vicon_name + "_position")
-        self.robot.add_ros_and_trace(self.client_name, self.robot_vicon_name + "_velocity_body")
-        self.robot.add_ros_and_trace(self.client_name, self.robot_vicon_name + "_velocity_world")
+        self.robot.add_ros_and_trace(self.vicon_client.name, self.robot_vicon_name + "_position")
+        self.robot.add_ros_and_trace(self.vicon_client.name, self.robot_vicon_name + "_velocity_body")
+        self.robot.add_ros_and_trace(self.vicon_client.name, self.robot_vicon_name + "_velocity_world")
 
         self.com_imp_ctrl = ComImpedanceControl(EntityName)
 
-    def compute_torques(self, Kp, des_pos, Kd, des_vel, des_fff):
+        self._biased_base_position = None
+        self._biased_base_velocity = None
 
+        self.vicon_offset = constVector([0., 0., 0.,])
 
-        self.base_pos_xyz = selec_vector(self.vicon_client.signal(self.robot_vicon_name + "_position"),
-                                        0, 3, "base_pos")
+    def init_robot_properties(self):
+        self.robot_vicon_name = "quadruped"
+        self.robot_mass = [2.17784, 2.17784, 2.17784]
```

Comment:
Why 3 data?

Reply:
Because the robot_mass is assumed to be a vector of 3.